src/main/scala/metadata.scala

```diff
@@ -0,0 +1,83 @@
+package suffuse
+
+/** AKey is an attribute key: a Metadata map key which carries
+ *  both the type of the corresponding value, and a value to
+ *  provide when there is no value in the map. This as opposed
+ *  to wrapping everything in sight in Option.
+ */
+final class AKey[A](description: String, empty: => A) {
+  def emptyValue: A = empty
```

Comment:
I'm struggling with this one. I never liked it when the absence of something was hidden with a default value. In this location however I can see the benefit of having it. If someone, with a gun to your head, demands a value for a given key, you are able to supply it.

What I'm unsure of however, is if we have any guys with guns in this department. Linux for example accepts not supplying `stats` in `getattr` and displays it to the user as `?`. For example I encountered the following:

```
eecolor@T450:~/test-mnt$ ls -al
ls: cannot access test.yaml: No such file or directory
total 16
drwxrwxr-x  1 eecolor eecolor 12288 jul 26  48174 .
drwxr-xr-x 46 eecolor eecolor  4096 mrt 13 12:36 ..
??????????  ? ?       ?           ?            ? test.yaml
```